cmd/charm/charmcmd/cmd.go

```diff
@@ -181,8 +181,8 @@ func addAuthFlag(f *gnuflag.FlagSet, s *string) {
 
 // addChannelFlag adds the -c (--channel) flags to the given flag set.
 func addChannelFlag(f *gnuflag.FlagSet, s *string) {
-	f.StringVar(s, "c", string(params.StableChannel), "the channel the charm or bundle is assigned to")
-	f.StringVar(s, "channel", string(params.StableChannel), "")
+	f.StringVar(s, "c", "", fmt.Sprintf("the channel the charm or bundle is assigned to (%s|%s|%s)", params.StableChannel, params.DevelopmentChannel, params.UnpublishedChannel))
```

Comment:
This is general for all commands using channels flag, the help on publish state that you can only use development or stable